### PR TITLE
topology: hdmi updates

### PR DESCRIPTION
**Pull request by plbossart**
rename hda topology file to use name expected by the kernel
add topology file to HDMI-only (no external codec) - credits to @ranj063 for editing this file, I only added and committed it.

Tested on Up2 with the kernel PR at https://github.com/thesofproject/linux/pull/624, which picks the right topology file depending on the number of HDaudio codecs detected.

Note that the devices start at one, this is done on purpose so that the HDMI devices remain at the same index between Up2 with PCM512x and Up2 without PCM512x.

### CODE REVIEW

File: tools/topology/sof-hda-generic-idisp.m4 [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#
+# Topology for generic Apollolake UP^2 with pcm512x codec and HDMI.
```
Comment (ranj063) [inline tools/topology/sof-hda-generic-idisp.m4]: @plbossart only minor but this comment needs to be amended.

File: tools/topology/sof-hda-generic-idisp.m4 [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`ssp.m4')
```
Comment (ranj063) [inline tools/topology/sof-hda-generic-idisp.m4]: @plbossart And we dont need to include ssp.m4

File: tools/topology/sof-hda-generic-idisp.m4 [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Apollolake DSP configuration
+include(`platform/intel/bxt.m4')
```
Comment (ranj063) [inline tools/topology/sof-hda-generic-idisp.m4]: and I thinking this too but not sure.

File: tools/topology/sof-hda-generic-idisp.m4 (hunk 1 above)
Comment (plbossart) [reply tools/topology/sof-hda-generic-idisp.m4]: ok

File: tools/topology/sof-hda-generic-idisp.m4 (hunk 2 above)
Comment (plbossart) [reply tools/topology/sof-hda-generic-idisp.m4]: ok

File: tools/topology/sof-hda-generic-idisp.m4 (hunk 3 above)
Comment (plbossart) [reply tools/topology/sof-hda-generic-idisp.m4]: not sure how, this is the same in the hda-generic part. and while it's a specific case it's not really broken in the sense that the same settings are used for HDMI in all platforms anyways.

**Comment by plbossart** [comment]
pushed update to add support for CMake and address @ranj063 comments
I have no idea what the virtual widgets mean so that will have to be done in a follow-up PR.

**Comment by RanderWang** [comment]
@plbossart @lgirdwood  I renamed hda-generic to hda-intel-common because there is a special change for intel platforms, so Liam advised me to rename it to hda-intel-common. I just want to change it in kernel.

**Comment by lgirdwood** [comment]
@plbossart @RanderWang if hdmi-generic-idisp.m4 has Intel only changes then it will need to use "intel" in the naming like the previous change. Please confirm before it's merged.

**Comment by plbossart** [comment]
> @plbossart <https://github.com/plbossart> @RanderWang 
> <https://github.com/RanderWang> if hdmi-generic-idisp.m4 has Intel 
> only changes then it will need to use "intel" in the naming like the 
> previous change. Please confirm before it's merged.
>
The kernel has been using 'sof-hda-generic.tplg' since August 22. I see 
no reason why the .m4 name would need to be different?

**Comment by lgirdwood** [comment]
> @plbossart <https://github.com/plbossart> @RanderWang <https://github.com/RanderWang> if hdmi-generic-idisp.m4 has Intel only changes then it will need to use "intel" in the naming like the previous change. Please confirm before it's merged.
> The kernel has been using 'sof-hda-generic.tplg' since August 22. I see no reason why the .m4 name would need to be different?

Intel only changes were made to this file to fix a bug that could be detrimental to non Intel uses, hence the generic names no longer applies.

**Comment by plbossart** [comment]
On 2/20/19 10:55 AM, Liam Girdwood wrote:
>
>     @plbossart <https://github.com/plbossart>
>     https://github.com/plbossart @RanderWang
>     <https://github.com/RanderWang> https://github.com/RanderWang if
>     hdmi-generic-idisp.m4 has Intel only changes then it will need to
>     use "intel" in the naming like the previous change. Please confirm
>     before it's merged.
>     The kernel has been using 'sof-hda-generic.tplg' since August 22.
>     I see no reason why the .m4 name would need to be different?
>
> Intel only changes were made to this file to fix a bug that could be 
> detrimental to non Intel uses, hence the generic names no longer applies.
>
What are you referring to? I don't see any Intel-specific changes in 
this file, and since there's been 6 months of use with this file used 
there is a kernel legacy to take care of...I also don't see who would 
use SOF with a non-Intel HDAudio interface without having a need to 
specific a different DAI type. The name of the topology file would be 
the least of the worries an implementer would need to take care of.

**Comment by RanderWang** [comment]
@plbossart  it is https://github.com/thesofproject/sof/commit/a5d32b4c3a0551a7e42e0b2af0b0fb17ad6f2079#diff-5421b23f2d0fae66995b660b708d20ab

**Comment by plbossart** [comment]
On 2/21/19 2:16 AM, RanderWang wrote:
>
> @plbossart <https://github.com/plbossart> it is 
> a5d32b4#diff-5421b23f2d0fae66995b660b708d20ab 
> <https://github.com/thesofproject/sof/commit/a5d32b4c3a0551a7e42e0b2af0b0fb17ad6f2079#diff-5421b23f2d0fae66995b660b708d20ab>
>
I don't see how this fix is Intel specific. It rather looks like a 
mismatch between ALSA and HDAudio representations.

Seriously, we are splitting hair about a file name that's already been 
used for 6 months. Come on.

**Comment by RanderWang** [comment]
> On 2/21/19 2:16 AM, RanderWang wrote: @plbossart <https://github.com/plbossart> it is [a5d32b4](https://github.com/thesofproject/sof/commit/a5d32b4c3a0551a7e42e0b2af0b0fb17ad6f2079)#diff-5421b23f2d0fae66995b660b708d20ab <[a5d32b4#diff-5421b23f2d0fae66995b660b708d20ab](https://github.com/thesofproject/sof/commit/a5d32b4c3a0551a7e42e0b2af0b0fb17ad6f2079#diff-5421b23f2d0fae66995b660b708d20ab)>
> I don't see how this fix is Intel specific. It rather looks like a mismatch between ALSA and HDAudio representations.  

yes, I agree with this idea

**Comment by plbossart** [comment]
Updated code to fix warnings. Yet another miss in Github that you don't get a notification when a PR no longer applies due to changes upstream.

**Pull request merged**